internal/services/databricks/databricks_root_dbfs_customer_managed_key_resource.go

```diff
@@ -67,6 +67,13 @@ func resourceDatabricksWorkspaceRootDbfsCustomerManagedKey() *pluginsdk.Resource
 				Required:     true,
 				ValidateFunc: keyVaultValidate.KeyVaultChildID,
 			},
+
+			// added to support cross subscription cmk's
+			"key_vault_id": {
```

Comment:
So i am wondering if there is a way to solve this with out adding a new property?

key_vault_key.id is the data plane id which is a uri, however we have key_vault_key.resource[_versionless]_id which is the resource manager ID which should contain the subscription.

key_vault_key_id should likely already expect the RM ID (vs key_vault_key_uri which is the data plane URI)

i imagine we might be able to support both for the property, however it would break import as we wouldn't be able to know which format to write on an import - would it make sense to just do a breaking change here?

Reply:
Using a RM ID isn't possible here since you can't get the subscription from the URL and the API only contains the key vault key URL...